In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import os

import dhs_util
from dhs_util import *

os.chdir('/Users/owner/Documents/WC_DS/')
df = pd.read_csv('dhs_service_records_synthesized_final.csv')
df = dhs_preprocessing(df)
df, service_map = add_service_label(df)
df = add_age_bin(df)

recipient = get_recipient_attribute(df)

In [9]:
df

id       date                                            service  \
0          1626 2021-01-31  Children_Attending_Early_Childhood_Programs_Ma...   
1          1626 2021-02-28  Children_Attending_Early_Childhood_Programs_Ma...   
2          1626 2021-03-31  Children_Attending_Early_Childhood_Programs_Ma...   
3          1626 2021-04-30  Children_Attending_Early_Childhood_Programs_Ma...   
4          1626 2021-05-31  Children_Attending_Early_Childhood_Programs_Ma...   
...         ...        ...                                                ...   
7116129  530832 2021-06-30                                           Suicides   
7116130  532121 2021-04-30                                           Suicides   
7116131  533991 2021-11-30                                           Suicides   
7116132  534127 2021-07-31                                           Suicides   
7116133  535156 2021-06-30                                           Suicides   

         age    gender     race              ethnicity  \
0          5    1~Male  1~White             99~Unknown   
1          5    1~Male  1~White             99~Unknown   
2          5    1~Male  1~White             99~Unknown   
3          5    1~Male  1~White             99~Unknown   
4          5    1~Male  1~White             99~Unknown   
...      ...       ...      ...                    ...   
7116129   17    1~Male  1~White             99~Unknown   
7116130   22  2~Female  1~White             99~Unknown   
7116131   16    1~Male  1~White             99~Unknown   
7116132   19    1~Male  1~White  2~Not Hispanic/Latinx   
7116133   24    1~Male  1~White             99~Unknown   

                        marital                      education     month serv  \
0                    99~Unknown  9-12~High School (grade 9-12)   January  S01   
1                    99~Unknown  9-12~High School (grade 9-12)  February  S01   
2                    99~Unknown  9-12~High School (grade 9-12)     March  S01   
3                    99~Unknown  9-12~High School (grade 9-12)     April  S01   
4                    99~Unknown  9-12~High School (grade 9-12)       May  S01   
...                         ...                            ...       ...  ...   
7116129              99~Unknown                     99~Unknown      June  S22   
7116130  1~Single-Never Married  9-12~High School (grade 9-12)     April  S22   
7116131  1~Single-Never Married  9-12~High School (grade 9-12)  November  S22   
7116132               4~Widowed           GRAD~Graduate Degree      July  S22   
7116133  1~Single-Never Married  9-12~High School (grade 9-12)      June  S22   

        age_bin  
0           0-9  
1           0-9  
2           0-9  
3           0-9  
4           0-9  
...         ...  
7116129   10-19  
7116130   20-29  
7116131   10-19  
7116132   10-19  
7116133   20-29  

[7116134 rows x 12 columns]

In [37]:
df_1 = df.query("serv == 'S14'")
df_2 = df.query("serv != 'S14'")

In [84]:
age_lower = 24
age_upper = 55
id_cohort_age = df.query("age > @age_lower and age <= @age_upper").id.unique()
id_cohort_race = df.query("race != '99~Unknown'").id.unique()
recipient_1 = get_recipient_attribute(df_1)
recipient_2 = get_recipient_attribute(df_2)

In [53]:
recipient_1

id  num_service  distinct_service first_date  last_date  \
0           17            1                 1 2021-10-31 2021-10-31   
1           23            3                 1 2021-05-31 2021-12-31   
4           26            2                 1 2021-03-31 2021-04-30   
6           35            4                 1 2021-02-28 2021-07-31   
10          38            1                 1 2021-12-31 2021-12-31   
...        ...          ...               ...        ...        ...   
324215  535566            7                 1 2021-01-31 2021-12-31   
324222  535576            4                 1 2021-04-30 2021-09-30   
324226  535604            5                 1 2021-01-31 2021-11-30   
324231  535605            1                 1 2021-08-31 2021-08-31   
324232  535606            9                 1 2021-01-31 2021-11-30   

        num_month  distinct_month  age    gender                      race  \
0               1               1   37    1~Male                   1~White   
1               3               3   47  2~Female                   1~White   
4               2               2   36    1~Male                   1~White   
6               4               4   56  2~Female                   1~White   
10              1               1   29    1~Male                   1~White   
...           ...             ...  ...       ...                       ...   
324215          7               7   50    1~Male                   1~White   
324222          4               4   51  2~Female  2~Black/African American   
324226          5               5   27    1~Male                   1~White   
324231          1               1   20  2~Female  2~Black/African American   
324232          9               9   75  2~Female                   1~White   

                    ethnicity                 marital  \
0       2~Not Hispanic/Latinx  1~Single-Never Married   
1       2~Not Hispanic/Latinx  1~Single-Never Married   
4       2~Not Hispanic/Latinx  1~Single-Never Married   
6       2~Not Hispanic/Latinx  1~Single-Never Married   
10      2~Not Hispanic/Latinx  1~Single-Never Married   
...                       ...                     ...   
324215  2~Not Hispanic/Latinx  1~Single-Never Married   
324222  2~Not Hispanic/Latinx               2~Married   
324226  2~Not Hispanic/Latinx  1~Single-Never Married   
324231  2~Not Hispanic/Latinx  1~Single-Never Married   
324232  2~Not Hispanic/Latinx  1~Single-Never Married   

                            education serv age_bin  
0              UNDERGRAD~Some College  S14   30-39  
1                          99~Unknown  S14   40-49  
4       9-12~High School (grade 9-12)  S14   30-39  
6       9-12~High School (grade 9-12)  S14   50-59  
10                         99~Unknown  S14   20-29  
...                               ...  ...     ...  
324215  9-12~High School (grade 9-12)  S14   50-59  
324222  9-12~High School (grade 9-12)  S14   50-59  
324226  9-12~High School (grade 9-12)  S14   20-29  
324231  9-12~High School (grade 9-12)  S14   20-29  
324232  9-12~High School (grade 9-12)  S14   70-79  

[55263 rows x 15 columns]

In [55]:
recipient_2

id  num_service  distinct_service first_date  last_date  \
0             1           12                 1 2021-01-31 2021-12-31   
12            2            1                 1 2021-02-28 2021-02-28   
13            3           11                 1 2021-02-28 2021-12-31   
24            4            4                 1 2021-03-31 2021-12-31   
28            5           12                 1 2021-01-31 2021-12-31   
...         ...          ...               ...        ...        ...   
6791797  535604           17                 2 2021-01-31 2021-12-31   
6791814  535605           13                 2 2021-01-31 2021-12-31   
6791827  535606           12                 2 2021-01-31 2021-12-31   
6791839  535607           42                 5 2021-01-31 2021-12-31   
6791881  535608           12                 1 2021-01-31 2021-12-31   

         num_month  distinct_month  age    gender                      race  \
0               12              12   13    1~Male                99~Unknown   
12               1               1   70  2~Female                   1~White   
13              11              11   29  2~Female                   1~White   
24               4               4    4  2~Female                   1~White   
28              12              12   40    1~Male  2~Black/African American   
...            ...             ...  ...       ...                       ...   
6791797         17              12   27    1~Male                   1~White   
6791814         13              12   20  2~Female  2~Black/African American   
6791827         12               9   75  2~Female                   1~White   
6791839         42              12   48  2~Female  2~Black/African American   
6791881         12              12    3  2~Female                99~Unknown   

                     ethnicity                 marital  \
0        2~Not Hispanic/Latinx              99~Unknown   
12       2~Not Hispanic/Latinx               2~Married   
13       2~Not Hispanic/Latinx               2~Married   
24       2~Not Hispanic/Latinx  1~Single-Never Married   
28       2~Not Hispanic/Latinx               2~Married   
...                        ...                     ...   
6791797  2~Not Hispanic/Latinx  1~Single-Never Married   
6791814  2~Not Hispanic/Latinx  1~Single-Never Married   
6791827  2~Not Hispanic/Latinx  1~Single-Never Married   
6791839  2~Not Hispanic/Latinx  1~Single-Never Married   
6791881             99~Unknown  1~Single-Never Married   

                                     education serv age_bin  
0        GED~High School Diploma/GED Completed  S12   10-19  
12                                  99~Unknown  S12   70-79  
13               9-12~High School (grade 9-12)  S12   20-29  
24               9-12~High School (grade 9-12)  S12     0-9  
28                      UNDERGRAD~Some College  S12   40-49  
...                                        ...  ...     ...  
6791797          9-12~High School (grade 9-12)  S09   20-29  
6791814          9-12~High School (grade 9-12)  S09   20-29  
6791827          9-12~High School (grade 9-12)  S09   70-79  
6791839                 UNDERGRAD~Some College  S06   40-49  
6791881          9-12~High School (grade 9-12)  S12     0-9  

[533799 rows x 15 columns]

In [90]:
id_white_df_1 = df_1.query("race == '1~White'")
id_minority_df_1 = df_1.query("race != '1~White'")
id_minority_df_1 = id_minority_df_1.query("race != '99~Unknown'")
id_white_df_2 = df_2.query("race == '1~White'")
id_minority_df_2 = df_2.query("race != '1~White'")
id_minority_df_2 = id_minority_df_2.query("race != '99~Unknown'")

In [104]:
id_white_df_1.nunique()

id           36801
date            12
service          1
age            106
gender           3
race             1
ethnicity        3
marital          7
education        8
month           12
serv             1
age_bin         11
dtype: int64

In [110]:
id_white_df_2.nunique()

id           334635
date             12
service          21
age             109
gender            3
race              1
ethnicity         3
marital           7
education         8
month            12
serv             21
age_bin          11
dtype: int64

In [114]:
perc_white_using_S14 = 36801/(36801+334635)
perc_white_using_S14

0.09907763383193875

In [116]:
id_minority_df_1.nunique()

id           17417
date            12
service          1
age            103
gender           3
race             5
ethnicity        3
marital          7
education        8
month           12
serv             1
age_bin         11
dtype: int64

In [118]:
id_minority_df_2.nunique()

id           177875
date             12
service          21
age             110
gender            3
race              5
ethnicity         3
marital           7
education         8
month            12
serv             21
age_bin          11
dtype: int64

In [120]:
perc_minority_using_S14 = 17417/(17417+177875)
perc_minority_using_S14

0.08918440079470741

### Conclusions
We found that white residents of Allegheny County were more likely to use mental health services, as 9.9% used S14 compared to the 8.9% of minority residents who did so. This could mean that Alleghany County is not making mental health services as easily accessible for minority residents.